python/dgl/storages/pytorch_tensor.py

```diff
@@ -5,21 +5,22 @@
 from .tensor import BaseTensorStorage
 from ..utils import gather_pinned_tensor_rows
 
-def _fetch_cpu(indices, tensor, feature_shape, device, pin_memory):
+def _fetch_cpu(indices, tensor, feature_shape, device, pin_memory, **kwargs):
     result = torch.empty(
         indices.shape[0], *feature_shape, dtype=tensor.dtype,
         pin_memory=pin_memory)
     torch.index_select(tensor, 0, indices, out=result)
-    result = result.to(device, non_blocking=True)
+    kwargs['non_blocking'] = True
```

Comment:
Should this be `kwargs['non_blocking'] = pin_memory`, since otherwise I believe torch >=1.9 will first copy the tensor to a pinned tensor before beginning the async copy?